src/bundle/Routing/ExpressionLanguage/ContentTypeHeaderMatcherFactory.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Bundle\Rest\Routing\ExpressionLanguage;
+
+use Closure;
+use Ibexa\Contracts\Rest\Input\ParsingDispatcher;
+use Symfony\Component\HttpFoundation\Request;
+
+final readonly class ContentTypeHeaderMatcherFactory
+{
+    public function __construct(
+        private ParsingDispatcher $parsingDispatcher
+    ) {
+    }
+
+    public function __invoke(): Closure
+    {
+        return function (Request $request, ?string $contentTypeHeaderToMatch): bool {
```

Comment:
Does it make sense to allow passing `$contentTypeHeaderToMatch === null`?

Reply:
Not really, it won't even be null :)